### ISSUE TITLE
Protologger Galaxy error

### ISSUE THREAD

**Issue by cpavloud** [thh32/Protologger#9]
Hi!

I cannot send this via Galaxy, so I am posting my issue here. 

I am trying to run Protologger via Galaxy and I get the job gets cancelled. 
I am getting this error report:

usearch v5.2.32 (32-bit, 2.0Gb avail / 528Gb RAM)
(C) Copyright 2010-11 Robert C. Edgar, all rights reserved.
License: th.tomhitch@gmail.com

00:00  20Mb    0.1% Reading /DATA/galaxy-run/galaxy/database/files/013/dataset_13419.dat
00:00  20Mb    0.1% Reading /DATA/galaxy-run/galaxy/database/files/013/dataset_13419.dat
00:00  33Mb  100.0% Reading /DATA/galaxy-run/galaxy/database/files/013/dataset_13419.dat
00:00  33Mb 1 sequences                                                                 
00:00  35Mb    0.1% Reading /DATA/galaxy-run/galaxy/tools/protologger/bin/16S-SILVA/LTP-DB/LTPs132_SSU_compressed.fasta, 0 seqs
WARNING: Invalid FASTA file '/DATA/galaxy-run/galaxy/tools/protologger/bin/16S-SILVA/LTP-DB/LTPs132_SSU_compressed.fasta', non-letter ' ' in sequence >FJ611848-1-1348-1348bp-rna-Erwinia gerundensis-Erwiniaceae

00:01 103Mb   20.2% Reading /DATA/galaxy-run/galaxy/tools/protologger/bin/16S-SILVA/LTP-DB/LTPs132_SSU_compressed.fasta, 2788 seqs
00:01 316Mb  100.0% Reading /DATA/galaxy-run/galaxy/tools/protologger/bin/16S-SILVA/LTP-DB/LTPs132_SSU_compressed.fasta, 13.9k seqs
00:01 300Mb  100.0% Chimera search /DATA/galaxy-run/galaxy/database/files/013/dataset_13419.dat, 0/1 found (0.0%)                  

*******************************************************************************
 [CheckM - tree] Placing bins in reference genome tree.
*******************************************************************************

  Identifying marker genes in 1 bins with 1 threads:
    Finished processing 0 of 1 (0.00%) bins.
    Finished processing 1 of 1 (100.00%) bins.
  Saving HMM info to file.

  Calculating genome statistics for 1 bins with 1 threads:
    Finished processing 0 of 1 (0.00%) bins.
    Finished processing 1 of 1 (100.00%) bins.

  Extracting marker genes to align.
  Parsing HMM hits to marker genes:
    Finished parsing hits for 1 of 1 (100.00%) bins.
  Extracting 43 HMMs with 1 threads:
    Finished extracting 0 of 43 (0.00%) HMMs.
    Finished extracting 1 of 43 (2.33%) HMMs.
    Finished extracting 2 of 43 (4.65%) HMMs.
    Finished extracting 3 of 43 (6.98%) HMMs.
    Finished extracting 4 of 43 (9.30%) HMMs.
    Finished extracting 5 of 43 (11.63%) HMMs.
    Finished extracting 6 of 43 (13.95%) HMMs.
    Finished extracting 7 of 43 (16.28%) HMMs.
    Finished extracting 8 of 43 (18.60%) HMMs.
    Finished extracting 9 of 43 (20.93%) HMMs.
    Finished extracting 10 of 43 (23.26%) HMMs.
    Finished extracting 11 of 43 (25.58%) HMMs.
    Finished extracting 12 of 43 (27.91%) HMMs.
    Finished extracting 13 of 43 (30.23%) HMMs.
    Finished extracting 14 of 43 (32.56%) HMMs.
    Finished extracting 15 of 43 (34.88%) HMMs.
    Finished extracting 16 of 43 (37.21%) HMMs.
    Finished extracting 17 of 43 (39.53%) HMMs.
    Finished extracting 18 of 43 (41.86%) HMMs.
    Finished extracting 19 of 43 (44.19%) HMMs.
    Finished extracting 20 of 43 (46.51%) HMMs.
    Finished extracting 21 of 43 (48.84%) HMMs.
    Finished extracting 22 of 43 (51.16%) HMMs.
    Finished extracting 23 of 43 (53.49%) HMMs.
    Finished extracting 24 of 43 (55.81%) HMMs.
    Finished extracting 25 of 43 (58.14%) HMMs.
    Finished extracting 26 of 43 (60.47%) HMMs.
    Finished extracting 27 of 43 (62.79%) HMMs.
    Finished extracting 28 of 43 (65.12%) HMMs.
    Finished extracting 29 of 43 (67.44%) HMMs.
    Finished extracting 30 of 43 (69.77%) HMMs.
    Finished extracting 31 of 43 (72.09%) HMMs.
    Finished extracting 32 of 43 (74.42%) HMMs.
    Finished extracting 33 of 43 (76.74%) HMMs.
    Finished extracting 34 of 43 (79.07%) HMMs.
    Finished extracting 35 of 43 (81.40%) HMMs.
    Finished extracting 36 of 43 (83.72%) HMMs.
    Finished extracting 37 of 43 (86.05%) HMMs.
    Finished extracting 38 of 43 (88.37%) HMMs.
    Finished extracting 39 of 43 (90.70%) HMMs.
    Finished extracting 40 of 43 (93.02%) HMMs.
    Finished extracting 41 of 43 (95.35%) HMMs.
    Finished extracting 42 of 43 (97.67%) HMMs.
    Finished extracting 43 of 43 (100.00%) HMMs.
  Aligning 43 marker genes with 1 threads:
    Finished aligning 0 of 43 (0.00%) marker genes.
    Finished aligning 1 of 43 (2.33%) marker genes.
    Finished aligning 2 of 43 (4.65%) marker genes.
    Finished aligning 3 of 43 (6.98%) marker genes.
    Finished aligning 4 of 43 (9.30%) marker genes.
    Finished aligning 5 of 43 (11.63%) marker genes.
    Finished aligning 6 of 43 (13.95%) marker genes.
    Finished aligning 7 of 43 (16.28%) marker genes.
    Finished aligning 8 of 43 (18.60%) marker genes.
    Finished aligning 9 of 43 (20.93%) marker genes.
    Finished aligning 10 of 43 (23.26%) marker genes.
    Finished aligning 11 of 43 (25.58%) marker genes.
    Finished aligning 12 of 43 (27.91%) marker genes.
    Finished aligning 13 of 43 (30.23%) marker genes.
    Finished aligning 14 of 43 (32.56%) marker genes.
    Finished aligning 15 of 43 (34.88%) marker genes.
    Finished aligning 16 of 43 (37.21%) marker genes.
    Finished aligning 17 of 43 (39.53%) marker genes.
    Finished aligning 18 of 43 (41.86%) marker genes.
    Finished aligning 19 of 43 (44.19%) marker genes.
    Finished aligning 20 of 43 (46.51%) marker genes.
    Finished aligning 21 of 43 (48.84%) marker genes.
    Finished aligning 22 of 43 (51.16%) marker genes.
    Finished aligning 23 of 43 (53.49%) marker genes.
    Finished aligning 24 of 43 (55.81%) marker genes.
    Finished aligning 25 of 43 (58.14%) marker genes.
    Finished aligning 26 of 43 (60.47%) marker genes.
    Finished aligning 27 of 43 (62.79%) marker genes.
    Finished aligning 28 of 43 (65.12%) marker genes.
    Finished aligning 29 of 43 (67.44%) marker genes.
    Finished aligning 30 of 43 (69.77%) marker genes.
    Finished aligning 31 of 43 (72.09%) marker genes.
    Finished aligning 32 of 43 (74.42%) marker genes.
    Finished aligning 33 of 43 (76.74%) marker genes.
    Finished aligning 34 of 43 (79.07%) marker genes.
    Finished aligning 35 of 43 (81.40%) marker genes.
    Finished aligning 36 of 43 (83.72%) marker genes.
    Finished aligning 37 of 43 (86.05%) marker genes.
    Finished aligning 38 of 43 (88.37%) marker genes.
    Finished aligning 39 of 43 (90.70%) marker genes.
    Finished aligning 40 of 43 (93.02%) marker genes.
    Finished aligning 41 of 43 (95.35%) marker genes.
    Finished aligning 42 of 43 (97.67%) marker genes.
    Finished aligning 43 of 43 (100.00%) marker genes.

  Reading marker alignment files.
  Concatenating alignments.
  Placing 1 bins into the genome tree with pplacer (be patient).

  { Current stage: 0:02:38.590 || Total: 0:02:38.590 }

*******************************************************************************
 [CheckM - lineage_set] Inferring lineage-specific marker sets.
*******************************************************************************

  Reading HMM info from file.
  Parsing HMM hits to marker genes:
    Finished parsing hits for 1 of 1 (100.00%) bins.

  Determining marker sets for each genome bin.

    Finished processing 1 of 1 (100.00%) bins (current: dataset_13418).

  Marker set written to: /DATA/galaxy-run/galaxy/tools/protologger/Output/jjlmtuce/CheckM_results/lineage.ms

  { Current stage: 0:00:00.592 || Total: 0:02:39.182 }

*******************************************************************************
 [CheckM - analyze] Identifying marker genes in bins.
*******************************************************************************

  Identifying marker genes in 1 bins with 1 threads:
    Finished processing 0 of 1 (0.00%) bins.
    Finished processing 1 of 1 (100.00%) bins.
  Saving HMM info to file.

  { Current stage: 0:03:52.259 || Total: 0:06:31.442 }

  Parsing HMM hits to marker genes:
    Finished parsing hits for 1 of 1 (100.00%) bins.
  Aligning marker genes with multiple hits in a single bin:
    Finished processing 0 of 1 (0.00%) bins.
    Finished processing 1 of 1 (100.00%) bins.

  { Current stage: 0:00:01.213 || Total: 0:06:32.655 }

  Calculating genome statistics for 1 bins with 1 threads:
    Finished processing 0 of 1 (0.00%) bins.
    Finished processing 1 of 1 (100.00%) bins.

  { Current stage: 0:00:00.286 || Total: 0:06:32.942 }

*******************************************************************************
 [CheckM - qa] Tabulating genome statistics.
*******************************************************************************

  Calculating AAI between multi-copy marker genes.

  Reading HMM info from file.
  Parsing HMM hits to marker genes:
    Finished parsing hits for 1 of 1 (100.00%) bins.

  { Current stage: 0:00:00.900 || Total: 0:06:33.843 }
**Traceback (most recent call last):
  File "/DATA/galaxy-run/galaxy/tools/protologger/protologger-0.98-GalaxyEdition.py", line 582, in <module>
    sim,total = Similarity(seq1,seq2)
NameError: name 'seq1' is not defined**

Do you have any idea how to solve this?
Thank you!

**Comment by thh32** [comment]
Thank you for sending us this error message, this is really strange! It seems to do all the quality checks and then fail at the sequence alignment step. Initially I can't think of what might cause this, but if youplease send me your input files (thitch@ukaachen.de), I will test the issue and try to resolve it.

**Comment by thh32** [comment]
Sorry for closing the issue, it is now reopened (done on phone)

**Comment by thh32** [comment]
Hey,

I have just tried and replicated your error. I then changed the name of your 16S rRNA file so the sequence name is just '>16S_rRNA' and this seems to have resolved the issue.

I believe it is an issue with the separators used in the original name which are incompatible with the HTSeq python module. I have expeciered this before as HTSeq sometimes takes the first part e.g. '16S_rRNA' instead of '16S_rRNA:nameofspecies', which means it is unable to match it to the full name which are in other, non HTSeq read files, such as the alignment output.

By changing this, your files ran through the Protologger server without issue.

Thank you again for bringing this to my attention, I will include a primary step in the next iteration of Protologger which renames the input straight away so we avoid this.

Thanks,
Tom
